pages/docs/deployment/migrate-v2-to-v3.mdx

```diff
@@ -0,0 +1,223 @@
+It accepts events from a Redis queue and ensures that they are eventually being upserted into Clickhouse.
+
+
+## SDK compatibility and API changes
+
+While we aim to keep our SDKs and APIs fully backwards compatible, we have to introduce backwards incompatible changes with our update to V3.
+Certain APIs in SDK versions below version 2.0.0 are not compatible with our new backend architecture.
+Please upgrade and benefit from many performance improvements or features such as [prompt caching](https://langfuse.com/changelog/2024-02-05-sdk-level-prompt-caching).
+
+**Upgrade options**:
+- Default SDK upgrade: We wrote documentation on how to upgrade from 1.x.x to 2.x.x ([Python](https://langfuse.com/docs/sdk/python/low-level-sdk#upgrading-from-v1xx-to-v2xx), [JS](https://langfuse.com/docs/sdk/typescript/guide#upgrade1to2)). For the JS SDK, we also have an upgrade path [from 2.x.x to 3.x.x](https://langfuse.com/docs/sdk/typescript/guide#upgrade2to3). The upgrade is straightforward and should not take much time.
+- Improved integrations: Since the first major version, we built many new ways to integrate your code with Langfuse such as [Decorators](https://langfuse.com/docs/sdk/python/decorators) for Python. We would recommend to check out our [quickstart](https://langfuse.com/docs/get-started) to see whether there is a more convenient integration available for you.
+
+**Background**: Langfuse V3 relies on an event driven backend architecture.
+This means, that we acknowledge HTTP requests from the SDKs, queue the HTTP bodies in the backend, and process them asynchronously.
+This allows us to scale the backend more easily and handle more requests without overloading the database.
+The SDKs below 2.0.0 send the events to our server and expect a synchronous response containing the database representation of the event.
+If you rely on this data and access it in the code, your SDK will break as of Nov. 11th, 2024 for the cloud version and post-upgrade for self-hosted versions.
+
+### API Changes
+
+#### POST /api/public/ingestion
+
+The `/api/public/ingestion` endpoint is now asynchronous.
+It will accept all events as they come in and queue them for processing before returning a 207 status code.
+This means that events will _not_ be available immediately after acceptance by the backend and instead will appear
+eventually in subsequent read requests.
+
+The individual events accepted a `metadata` property within their body of type `string | string[] | Record<string, unknown>`.
+Only the `Record` type is supported within our UI and endpoints to perform queries and filter events.
+Therefore, we enforce an object type for `metadata` going forward.
+All incoming events with `{ metadata: string | string[] }` will have their metadata mapped to an object with key `metadata`,
+i.e. `{ event: { body: { metadata: "test" } } }` will be transformed to `{ event: { body: { metadata: { metadata: "test" } } } }`.
+
+#### POST /api/public/scores
```

Comment:
i'd include who does not need to worry about this, everyone using sdks as they do not use this endpoint